src/base/bittorrent/torrenthandle.h

```diff
@@ -425,7 +419,7 @@ namespace BitTorrent
 
         // Until libtorrent provide an "old_name" field in `file_renamed_alert`
         // we will rely on this workaround to remove empty leftover folders
-        QHash<LTFileIndex, QVector<QString>> m_oldPath;
+        QHash<int, QVector<QString>> m_oldPath;
```

Comment:
I don't think this is better for maintenance. When we drop libt 1.1 support no one will remember the `int` type should be `file_index_t`.
Perhaps we really should consider dropping libt 1.1 support in master branch instead of storing the underlying type.

Reply:
>When we drop libt 1.1 support no one will remember the int type should be file_index_t.

It shouldn't be `file_index_t` here.
Let me remind you that the original way was to narrow the distribution of these "trendy" libtorrent-1.2 types across the qBittorrent code as much as possible (by limiting them inside implementation of `BitTorrent::*` classes). I don't even remember why this one was here.

Reply:
>Let me remind you that the original way was to narrow the distribution of these "trendy"

Thanks, but I remember that.

>It shouldn't be file_index_t here.
>...
>I don't even remember why this one was here.

It is inevitable, just check `m_oldPath` usage, you added `LTUnderlyingType<LTFileIndex>` almost everywhere to it!
I really don't think storing the underlying type for the sake of hiding the type details is a better move.

Also what is your stance on using the new types in libt 1.2 if we dropped libt 1.1 support?
Will you still convert it back to the underlying types?

Reply:
> is inevitable, just check m_oldPath usage, you added LTUnderlyingType<LTFileIndex> almost everywhere to it!

Yes. But they all are located inside abstraction layer.
If you let this types into the `BitTorrent::*` interfaces, then you will get such type casts through all the qBittorrent code. In addition, it will require the addition of other things (e.g. `qHash()` functions).
So Yes, I would prefer current way.

If you think otherwise, you can try to implement your idea (just for fun) after this PR is merged.

Reply:
>Yes. But they all are located inside abstraction layer.

No, you're reversing the priorities IMO. Instead of following/using the native types from libtorrent, you're making the underlying type first class citizen and the new type second class (in this case).

>If you let this types into the BitTorrent::* interfaces

It will never leak out, it is in `private` section of the class.

Reply:
>you're making the underlying type first class citizen and the new type second class (in this case).

No, I don't.
"First class citizens" remain the same as they have always been in qBittorrent. I use underlying type just as an intermediate conversion step.

>It will never leak out, it is in `private` section of the class.

Even if I considered leaving this option as valid, now it requires an appropriate qHash function to be defined. I don't think it's worth it. I would strongly prefer to leave it the way I've implemented it currently.

Reply:
>>you're making the underlying type first class citizen and the new type second class (in this case).
>
>No, I don't.

I still think you do at this exact code, you're storing the underlying type instead of the intended/native type.

>Even if I considered leaving this option as valid, now it requires an appropriate qHash function to be defined.

Then I would just add it, you add it when you need it, why is it so hard?
So far seems only 1 additional qHash is required, that is not very much isn't?

Reply:
>>Even if I considered leaving this option as valid, now it requires an appropriate qHash function to be defined.
>
>Then I would just add it, you add it when you need it, why is it so hard?

Hard or not, I had to spend a few "precious" hours to do it the right way. Besides, I wouldn't call it a portable solution. Do you still prefer this way?

Reply:
yeah I do and thank you for putting the time to implement it.